### PR TITLE
Make the passenv environment setting case sensitive

### PR DESCRIPTION
**Pull request by gnikonorov**
## Description
Make the `passenv` environment setting case sensitive. Before this change it was including all environment variables that matched in a case insensitive fashion ( `FOO` included `FOO`, `fOo`, and `foo` for example )

Closes https://github.com/tox-dev/tox/issues/1534

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [x] added/updated test(s)
- [x] updated/extended the documentation - Not applicable in this case 
- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`,`breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with "by :user:`<your username>`"
  * please use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files - by :user:`superuser`."
  * also see [examples](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

**Comment by asottile** [comment]
at least on posix this is incorrect, `A` and `a` are two separate environment variables:

```console
$ a=1 A=2 env | grep -i '^a='
A=2
a=1
```

**Comment by asottile** [comment]
on windows however, `a` and `A` are the same variable:

```
C:\Users\Anthony>set A=1

C:\Users\Anthony>set a=2

C:\Users\Anthony>echo %A% %a%
2 2
```

**Comment by gnikonorov** [comment]
> on windows however, `a` and `A` are the same variable:
> 
> ```
> C:\Users\Anthony>set A=1
> 
> C:\Users\Anthony>set a=2
> 
> C:\Users\Anthony>echo %A% %a%
> 2 2
> ```

Thank you @asottile. I forgot about Windows. I'll add an exclusion for it.

**Comment by asottile** [comment]
I'm 50/50 on whether we should change this, or whether we should just document it as case insensitive

**Comment by gnikonorov** [comment]
Since this is my first time opening a PR in this repo, I'll defer to more experienced committers. Thoughts @gaborbernat ( since we spoke on the linked issue ) ?

**Review by gaborbernat — CHANGES_REQUESTED** [review]
> FWIW I think it would be easy and clean enough to just uppercase `name` and `spec` if `platform.system() == "Windows":`

Yeah, on Windows we need to be the case insensitive.

**Comment by asottile** [comment]
On second thought, changing this behaviour is likely to break a lot of people [even tox itself](https://github.com/tox-dev/tox/blob/c5a10d730e562164b03f2669596e0168ba9b3aac/tox.ini#L25-L27)

I'm leaning more towards not changing the behaviour and just documenting the case insensitivity

**Comment by gnikonorov** [comment]
> On second thought, changing this behaviour is likely to break a lot of people [even tox itself](https://github.com/tox-dev/tox/blob/c5a10d730e562164b03f2669596e0168ba9b3aac/tox.ini#L25-L27)
> 
> 
> 
> I'm leaning more towards not changing the behaviour and just documenting the case insensitivity

At the very least, it sounds like a major version bump

**Comment by gaborbernat** [comment]
Ok. Let's not address here in the code. But document the behavior.

**Pull request closed without merging**
